client/src/app/ui/modules/media-upload-content/components/media-upload-content/media-upload-content.component.ts

```diff
@@ -82,7 +87,7 @@ export class MediaUploadContentComponent extends BaseUiComponent implements OnIn
     }
 
     public getDirectoryTitle(): string {
-        return this.currentDirectory?.title || ``;
+        return this.repo.getViewModel(this.directoryId)?.title || this.currentDirectory?.title || ``;
```

Comment:
Can this component be supplied with an id or a view model or are both needed? If the former, maybe it can be unified. If the latter, the id should be removed, since the view model also contains it.